src/input/masked-input.js

```diff
@@ -71,13 +71,16 @@ export default function MaskedInput({
         maskChar,
         ...(inputOverride.props || {}),
       },
-      style: {
-        ...(inputOverride.style || {}),
-      },
+      style:
+        typeof inputOverride.style === 'function'
+          ? inputOverride.style
+          : {
+              ...(inputOverride.style || {}),
+            },
     },
     ...restOverrides,
   };
-
+  // $FlowFixMe
```

Comment:
the code below will give no flow errors and also allows for component overrides, which was not supported before

```js

export default function MaskedInput({
  mask,
  maskChar,
  overrides: {Input: inputOverride, ...restOverrides} = {},
  ...restProps
}: MaskedInputPropsT) {
  let componentOverride = MaskOverride;
  let propsOverride = {};
  let styleOverride = {};

  if (typeof inputOverride === 'function') {
    componentOverride = inputOverride;
  } else if (typeof inputOverride === 'object') {
    componentOverride = inputOverride.component;
    propsOverride = inputOverride.props;
    styleOverride = inputOverride.style;
  }

  const nextOverrides = {
    Input: {
      component: componentOverride,
      props: {
        mask,
        maskChar,
        ...propsOverride,
      },
      style: styleOverride,
    },
    ...restOverrides,
  };

  return <Input {...restProps} overrides={nextOverrides} />;
}
```